scripts/gen_secret_patterns.py

```
@@ -0,0 +1,192 @@
+def _emit_rule(rule: dict[str, Any]) -> str:
+    rid = rule["id"]
+    pattern = _hoist_inline_flags(rid, rule["regex"])
+    keywords = tuple(sorted({k.lower() for k in rule.get("keywords") or ()}))
+    entropy = float(rule["entropy"]) if "entropy" in rule else 0.0
+
+    allowlist_parts: list[tuple[str, str]] = []
+    stopwords: set[str] = set()
+    for al in rule.get("allowlists") or ():
+        target = al.get("regexTarget", "match")
+        for r in al.get("regexes") or ():
+            allowlist_parts.append((target, _hoist_inline_flags(f"{rid}.allowlist", r)))
```

Ruff (0.15.12)

[warning] Use `list.extend` to create a transformed list

(PERF401)